### DISCUSSION TITLE
Repeater field & Repeatable order / position save to hasMany models

### DISCUSSION THREAD

**Discussion by couchlab-pl** [laravel/nova-issues#5846] [Ideas & Feature Requests]
- Laravel Version: 10
- Nova Version: 4.27.12
- PHP Version: 8.1

### Description:
Repeater field has the ability to change the order of individual items. How to save this value to the database, e.g. in the order or position field when using hasMany(). There is nothing in the documentation about this.

**Comment by LiamKarlMitchell** [comment]
Repeater field has up/down arrows to change the display order it would be great if we could link this to our own field/column.

For example when using with hasMany relation and upserting, it would be ideal to be able to have a custom order column that stores this position so that it can be used as a display order for the records when queried elsewhere by business logic. With the JSON I don't think it would be necessary as the array would already have the order, however someone might have a use for it to be stored there too.


Some challenges may be:

- Order should not have gaps
- Orders should not have duplicates duplicated for soft delete is probably ok.
- Order value should start from 1.
- Delete should update order value of other items below the currently deleted one.
- If using soft deletes then undelete must re-insert at the order given or next possible order if highest order is less than historical order value.

**Comment by mgralikowski** [comment]
At this moment it is possible to store an order of items - by skipping the `uniqueField()` modificator, we allow to remove all records and add them again, so the standard autoincrement column indicates in some way the order. The only drawback I see is the increase of ID value every time we change something, and generally, ID becomes not persistent.
